protected/humhub/modules/space/resources/js/humhub.space.setting.js

```diff
@@ -0,0 +1,22 @@
+
+humhub.module('space.settings', function(module, require, $) {
+    var modal = require('ui.modal');
+
+    var confirmVisibilityChange = function (evt) {
+        if (evt.$trigger.val() === '0') {
+            modal.confirm().then(function(confirmed) {
```

Comment:
We should use some specfic texts here: https://docs.humhub.org/docs/develop/javascript-modals/#modal-confirm

Reply:
You can define such texts in your AssetBundle as follows:

```
 public static function register($view)
    {
        $view->registerJsConfig('calendar.Calendar', [
            'confirmText' => [
                'header' => Yii::t(''SpaceModule.base'', 'Confirm space visibility change'),
                 'body' => Yii::t(''SpaceModule.base'', 'All content of this space will be changed to private group visibility!'),
            ]
        ]);
        return parent::register($view);
    }
```

In your module
```
modal.confirm(module.config.confirmText).then(...
```